cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/upgrade/AwsRdsUpgradeService.java

```diff
@@ -40,8 +40,10 @@ public void upgrade(AuthenticatedContext ac, DatabaseStack dbStack, Version targ
         AmazonRdsClient rdsClient = getAmazonRdsClient(ac);
         RdsInfo rdsInfo = getRdsInfo(dbInstanceIdentifier, rdsClient);
         validateIfRdsCanBeUpgraded(targetMajorVersion, rdsInfo);
-        upgadeRdsIfNeeded(targetMajorVersion, databaseServer, rdsClient, rdsInfo);
-        waitForRdsUpgrade(ac, databaseServer, rdsClient);
+        if (awsRdsUpgradeValidatorService.isRdsMajorVersionSmallerThanTarget(rdsInfo, targetMajorVersion)) {
```

Comment:
this seems to be quite counter intuitive for me: that

1. we run `validateIfRdsCanBeUpgraded` 
2. we run an extra validation `isRdsMajorVersionSmallerThanTarget`

What if `validateIfRdsCanBeUpgraded` would return a boolean that could drive this condition instead? It would be cleaner in my opinion

Reply:
I see. The difference is, that in case the validation fails, an exception is thrown. However, when the version is not smaller, than the targetMajorVersion, no exception should be thrown. So the two "validations" cannot be merged.
What I could do is to move method validateIfRdsCanBeUpgraded within the if(versionCheck) {}
Would it be cleaner to you?